### ISSUE TITLE
Should not lint on not locally used `@export`ed functions

### ISSUE THREAD

**Issue by radbasa** [Appsilon/box.linters#31]
## box.linters version

0.0.0.9006

## Source code to lint

```R
#' @export
public_function <- function() {
  private_function()
}

private_function <- function() {
  \"private function\"
}
```

## Lint command

```R
lintr::lint(code, linters = lintr::linters_with_defaults(defaults = box.linters::rhino_default_linters))
```

## Lint result

```
<text>:3:1: warning: [unused_declared_func_linter] Declared function unused.
public_function <- function() {
^~~~~~~~~~~~~~~
```

## Expected result

Should not lint on `@export`ed functions.
